token/services/ttx/endorse.go

```diff
@@ -745,6 +734,35 @@ func (s *endorseView) requestsToBeSigned() ([]*token.Transfer, error) {
 				res = append(res, transfer)
 			}
 		}
+		for _, sender := range transfer.ExtraSigners {
+			if _, err := s.tx.TokenService().SigService().GetSigner(sender); err == nil {
+				res = append(res, transfer)
+			}
+		}
 	}
 	return res, nil
 }
+
+func (s *endorseView) receiveTransaction(context view.Context) error {
+	if logger.IsEnabledFor(zapcore.DebugLevel) {
+		logger.Debugf("Receive transaction with envelope...")
+	}
+	// TODO: this might also happen multiple times because of the pseudonym. Avoid this by identity resolution at the sender
+	tx, err := ReceiveTransaction(context)
+	if err != nil {
+		return errors.Wrapf(err, "failed receiving transaction")
+	}
+	agent := metrics.Get(context)
```

Comment:
probably should open an issue about making this not a singleton.

Reply:
what do you mean here? agent is taken from the context.